### PR TITLE
194 add callback response

### LINKED ISSUE #194: Feature request: add callback function to ask API
Would be great if there was an optional callback function that would receive the response.
This would enable users to automate things with the response, e.x. take the commit message and use it in `git commit -m <response>`. Perhaps something simple like:
```
            require("CopilotChat").ask(input, {
               selection = ..,
               callback = function(response) {
                   vim.cmd("!git commit -m " .. response)
                }
            })
```

### PR DESCRIPTION
**Pull request by aweis89**
Fixes #194

### CODE REVIEW

File: lua/CopilotChat/config.lua [hunk 1]
```diff
@@ -63,6 +63,8 @@ local select = require('CopilotChat.select')
 ---@field selection nil|fun(source: CopilotChat.config.source):CopilotChat.config.selection?
 ---@field window CopilotChat.config.window?
 ---@field mappings CopilotChat.config.mappings?
+---@field callback fun(response: string)?
+
```
Comment (deathbeam) [inline lua/CopilotChat/config.lua:67]: Remove the space

File: lua/CopilotChat/config.lua [hunk 2]
```diff
@@ -63,6 +63,8 @@ local select = require('CopilotChat.select')
 ---@field selection nil|fun(source: CopilotChat.config.source):CopilotChat.config.selection?
 ---@field window CopilotChat.config.window?
 ---@field mappings CopilotChat.config.mappings?
+---@field callback fun(response: string)?
```
Comment (deathbeam) [inline lua/CopilotChat/config.lua:66]: Probs should define also default value in config as it is part of config spec (which includes also updating README). Also probs put it under selection so window and mappings stay at very bottom

**Review by deathbeam — CHANGES_REQUESTED** [review]
On top of this I think just small example in API README section would be nice (similar to the existing .ask examples there)

File: README.md [hunk 3]
```diff
@@ -168,7 +182,7 @@ actions.pick(actions.help_actions())
 
 -- Pick prompt actions
 actions.pick(actions.prompt_actions())
-```
+````
```
Comment (aweis89) [inline README.md:185]:
```suggestion
```
```

File: lua/CopilotChat/config.lua (hunk 2 above)
Comment (aweis89) [reply lua/CopilotChat/config.lua]: Not sure I follow, you mean an empty function that does nothing?
Is the optional nil `?` and handling that scenario not sufficient?

Comment (deathbeam) [reply lua/CopilotChat/config.lua]: Just callback = nil as default value yea, for ppl without LSP who are just looking at the config directly (and also for the README part where it shows just config)

File: README.md (hunk 3 above)
Comment (deathbeam) [inline README.md:185]: This should be just trple backtick no?

Comment (deathbeam) [reply README.md:185]: Oh nvm I see

File: README.md [hunk 4]
```diff
@@ -157,6 +157,20 @@ chat.ask("Explain how it works.", {
   selection = require("CopilotChat.select").buffer,
 })
 
+-- Ask a question and do something with the response
+local prompt = require("CopilotChat.config").prompts.Commit.prompt
```
Comment (deathbeam) [inline README.md:161]: This example can be just super simple, the git stuff can just be tip later maybe but I would simply just do `vim.print(response)` as example here, e.g super simple, without selection too

**Comment by deathbeam** [comment]
Adjusted the PR a bit with the changes i proposed

**Review by deathbeam — APPROVED** [review]

**Comment by deathbeam** [comment]
Hmm, let me try this I guess

@all-contributors add @aweis89 for code and doc.

**Pull request merged**